src/graph.rs

```diff
@@ -362,6 +362,35 @@ impl PyGraph {
         Ok(edge.index())
     }
 
+    pub fn add_edges_from(
+        &mut self,
+        obj_list: Vec<(usize, usize, PyObject)>,
```

Comment:
`py: Python` is only needed if the code is going to call python, it represents a GIL lock basically. Normally this is only to interact with an existing python object or to create a new python object. In this case it's not needed because PyO3 will handle the conversion of `list(int, int, object)` -> `Vec<(usize, usize, PyObject)` on the input and `Vec<usize>` -> `list(int)` on the return. So we don't need to acquire the GIL while this function is running.